### PR TITLE
Updated Swagger Files For SQL Managed Database and Managed Instance

### PR DESCRIPTION
**Pull request by ziwa-msft**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ x] The title of the PR is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ x] If applicable, the PR references the bug/issue that it fixes.
- [ x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ x] My spec meets the review criteria:
  - [ x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
Handling a bunch of network PRs right now

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+          "Creates a new managed database using point in time restore": {
+            "$ref": "./examples/ManagedDatabaseCreatePointInTimeRestore.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ManagedDatabases"
+        ],
+        "description": "Deletes the managed database.",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json]: Deletes ***a*** managed database

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+        "collation": {
+          "description": "Collation of the managed database.",
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "status": {
+          "description": "Status for the database.",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json]: Status ***of*** the database

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstances.json [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances": {
+      "get": {
+        "tags": [
+          "ManagedInstances"
+        ],
+        "description": "Gets a list of managed instances in a resource group.",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstances.json]: does this actually work?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstances.json [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/managedInstances": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedInstances.json]: Please test that this actually works.

**Review by jaredmoo — COMMENTED** [review]
Minor comments but looks good.

**Review by jaredmoo — APPROVED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceListByResourceGroup.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          }, {
+            "location" : "onebox",
+            "id" : "/subscriptions/20d7082a-0fc7-4468-82bd-542694d5042b/resourceGroups/Test1/providers/Microsoft.Sql/managedInstances/testcl2",
+            "name" : "testcl2",
+            "type": "Microsoft.Sql/managedInstances",
+            "sku" : {
+              "name" : "CLS15",
+              "tier" : "Standard",
+              "capacity": 22,
+              "vCores": 24,
```
Comment (jianghaolu) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceListByResourceGroup.json]: vCores, storageSizeInGB and licenseType should be in "properties".

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceListByResourceGroup.json (hunk 5 above)
Comment (jugeorge) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceListByResourceGroup.json]: Yes, sorry. We caught this and got the change checked into dsmain. Will update when we've hit production and are ready for check-in. Thank you.

Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/ManagedInstanceListByResourceGroup.json]: Is this just a swagger generating issue, or is it a production service issue?

**Comment by ziwa-msft** [comment]
Addressed comments

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Signed off from SDK side.

File: specification/sql/resource-manager/readme.md [hunk 6]
```diff
@@ -182,6 +184,7 @@ input-file:
 - Microsoft.Sql/preview/2015-05-01-preview/virtualNetworkRules.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: add 2015\managedInstances.json to package-2017

File: specification/sql/resource-manager/readme.md [hunk 7]
```diff
@@ -63,6 +63,7 @@ input-file:
 - Microsoft.Sql/preview/2015-05-01-preview/databaseAutomaticTuning.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Please merge from master and add managedInstances and managedDatabase to ALL package-composite-v* sections.

File: specification/sql/resource-manager/readme.md [hunk 8]
```diff
@@ -182,6 +184,7 @@ input-file:
 - Microsoft.Sql/preview/2015-05-01-preview/virtualNetworkRules.json
 - Microsoft.Sql/preview/2017-03-01-preview/cancelOperations.json
 - Microsoft.Sql/preview/2017-03-01-preview/dataWarehouseUserActivities.json
+- Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json
 - Microsoft.Sql/preview/2017-03-01-preview/renameDatabase.json
 - Microsoft.Sql/preview/2017-03-01-preview/serverAutomaticTuning.json
 - Microsoft.Sql/preview/2017-03-01-preview/serverDnsAliases.json
```
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: add 2015\managedInstances.json to package-pure-2015 and add 2017\managedDatabases.json to package-pure-2017

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Merge from master and update readme.md.

**Comment by jaredmoo** [comment]
- package-2017 should contain all files up to and including 2017.
 - package-2015 should contain all files up to and including 2015.
 - package-composite-v* should contain latest version of apis unless there is some backcompat isssue.
 - package-pure-2017 should contain all files of 2017 only.
 - package-pure-2015 should contain all files of 2015 only.

**Comment by ravbhatnagar** [comment]
Signing off from ARM

**Comment by jianghaolu** [comment]
Please let me know if you plan to address the comments from @jaredmoo. Thank you!

**Comment by jaredmoo** [comment]
Please do not merge without the readme.md changes I requested or else generated SDKs will not be right

**Comment by jugeorge** [comment]
We are planning on making the changes Jared requested, and we will also be adding InstanceFailoverGroup swagger in the readme because that is also ready. Waiting for the ARM manifest to be deployed (should be within a day or two). Our .NET SDKs are also ready and a PR will be started shortly.

**Comment by jugeorge** [comment]
Our ARM manifest has been deployed and we fixed the readme. Please merge as soon as possible.

**Pull request merged**